windows/vtpm_cold_add_remove/bitlocker_enable_test.yml

```diff
@@ -0,0 +1,109 @@
+- name: "Get VM firmware info for Windows client"
+  include_tasks: ../../common/vm_get_boot_info.yml
+  when: guest_os_product_type == 'client'
+
+- name: "Set fact of whether BitLocker Drive Encryption is enabled automatically"
+  ansible.builtin.set_fact:
+    win_bde_auto_enable: |-
+      {%- if guest_os_product_type == 'client' and guest_os_ansible_distribution_ver is version('10.0.26100.0', '>=') and vm_secureboot_enabled -%}true
+      {%- else -%}false
+      {%- endif -%}
+
+# BitLocker is not installed by default in Windows Server
+- name: "Check and install BitLocker feature in Windows Server"
+  when: guest_os_product_type == 'server'
+  block:
+    - name: "Check if BitLocker is installed"
+      include_tasks: ../utils/win_execute_cmd.yml
+      vars:
+        win_powershell_cmd: "(Get-WindowsFeature -Name BitLocker).InstallState"
+    - name: "Set fact of BitLocker install state"
+      ansible.builtin.set_fact:
+        win_srv_bitlocker: "{{ win_powershell_cmd_output.stdout_lines[0].strip() }}"
+
+    - name: "BitLocker is already installed"
+      ansible.builtin.debug: var=win_srv_bitlocker
+      when: win_srv_bitlocker in ['Installed', 'Enabled']
+
+    - name: "BitLocker is not installed"
+      when: win_srv_bitlocker not in ['Installed', 'Enabled']     
+      block:
+        - name: "Install BitLocker"
+          include_tasks: ../utils/win_execute_cmd.yml
+          vars:
+            win_powershell_cmd: "Install-WindowsFeature BitLocker"
+        - name: "Restart guest OS after BitLocker install"
+          include_tasks: ../utils/win_shutdown_restart.yml
+          vars:
+            set_win_power_state: "restart"
+
+- name: "Enable BitLocker on 'C:' drive"
+  include_tasks: ../utils/win_execute_cmd.yml
+  vars:
+    win_powershell_cmd: "Enable-BitLocker -MountPoint 'C:' -UsedSpaceOnly -TpmProtector -SkipHardwareTest"
+  when: not win_bde_auto_enable
+
+- name: "Retry to get 'C:' drive encryption state"
+  ansible.windows.win_shell: "manage-bde -status 'C:'"
+  register: get_encrypt_result
+  ignore_errors: true
+  delegate_to: "{{ vm_guest_ip }}"
+  ignore_unreachable: true
+  retries: 60
+  delay: 10
+  until:
+    - get_encrypt_result.stdout is defined
+    - "'Used Space Only Encrypted' in get_encrypt_result.stdout"
+    - "'100.0%' in get_encrypt_result.stdout"
+
+- name: "Check 'C:' drive encryption state"
+  ansible.builtin.assert:
+    that:
+      - get_encrypt_result.failed is defined
+      - not get_encrypt_result.failed
+    fail_msg: "Not get 'Used Space Only Encrypted' and/or '100.0%' percentage encrypted result after 600 seconds."
+
+- name: "Get 'C:' drive encryption key protector"
+  include_tasks: ../utils/win_execute_cmd.yml
+  vars:
+    win_powershell_cmd: "manage-bde -protectors -get 'C:'"
+    win_execute_cmd_ignore_error: true
+
+- name: "Set fact of the result of getting key protector"
+  ansible.builtin.set_fact:
+    get_key_protector_result: "{{ win_powershell_cmd_output }}"
```

Comment:
I think here can set `get_key_protector_result` to `win_powershell_cmd_output.stdout | default('')`, so that below when can check TPM not in get_key_protector_result